File: python/1046.py
initialTime, endTime = input().split()
initialTime = int(initialTime)
endTime = int(endTime)

totalTime = 0

if endTime < initialTime :
    totalTime = (24 - initialTime) + endTime
elif endTime == initialTime:
    totalTime = 24
else:
    totalTime = endTime - initialTime
    
print(f'O JOGO DUROU {totalTime} HORA(S)')

File: lua/1046.lua
local entry = io.read()
local t = {}
for v in string.gmatch(entry, '[^%s]+') do
    table.insert(t, tonumber(v))
end

local startTime = t[1]
local endTime = t[2]

local time = 0

if startTime >= endTime then
    time = 24 - startTime + endTime
else
    time = endTime - startTime
end

print('O JOGO DUROU '..time..' HORA(S)')